server/bin/test_data_importer.py

```diff
@@ -0,0 +1,706 @@
+
+        return self.post('users', user).json()
+
+    def create_role(self, role_name, permissions):
+        role = {
+            'name': role_name,
+            'permissions': permissions
+        }
+
+        return self.post('roles', role).json()
+
+    def add_role_to_user(self, username, role_name):
+        return self.post("roles/%s/users/%s" % (role_name, username)).json()
+
+    def create_content(self, owner_key, content_data):
+        content = safe_merge(content_data, {})
+
+        if self.hosted_test:
+            url = 'hostedtest/content'
+        else:
+            url = "owners/%s/content" % (owner_key)
+
+        return self.post(url, content).json()
+
+    def create_product(self, owner_key, product_data):
+        attributes = product_data['attributes'] if 'attributes' in product_data else {}
+        attributes['version'] = product_data['version'] if 'version' in product_data else "1.0"
+        attributes['variant'] = product_data['variant'] if 'variant' in product_data else "ALL"
+        attributes['arch'] = product_data['arch'] if 'arch' in product_data else "ALL"
+        attributes['type'] = product_data['type'] if 'type' in product_data else "SVC"
+
+        branding = []
+        content = product_data['content'] if 'content' in product_data else []
+        dependent_products = product_data['dependencies'] if 'dependencies' in product_data else []
+        relies_on = product_data['relies_on'] if 'relies_on' in product_data else []
+        provided_products = []
+        derived_product = None
+
+        # correct provided products and derived product
+        if 'provided_products' in product_data:
+            provided_products = [ { 'id': value } for value in product_data['provided_products'] ]
+
+        if 'derived_product_id' in product_data:
+            derived_product = { 'id': product_data['derived_product_id'] }
+
+        # Add branding to some of the products
+        # TODO: FIXME: couldn't we just do this in the test data instead?
+        if provided_products and 'OS' in product_data['name'] and 'type' in product_data and product_data['type'] == 'MKT':
+            branding.append({
+                'productId': provided_products[0]['id'],
+                'type': 'OS',
+                'name': "Branded %s" % (product_data['name'])
+            })
+
+        product = {
+            'id': product_data['id'],
+            'name': product_data['name'],
+            'multiplier': product_data['multiplier'] if 'multiplier' in product_data else 1,
+            'attributes': attributes,
+            'dependentProductIds': dependent_products,
+            'relies_on': relies_on,
+            'branding': branding,
+            'providedProducts': provided_products,
+            'derivedProduct': derived_product
+        }
+
+        # Determine URL and submit
+        if self.hosted_test:
+            url = 'hostedtest/products'
+        else:
+            url = "owners/%s/products" % (owner_key)
```

Comment:
Redundant parentheses.

Reply:
Will get fixed along with the other conversions of printf-style formatting.